src/mme-app/actionHandlers/attachActionHandlers.cpp

```diff
@@ -327,13 +340,31 @@ ActStatus ActionHandlers::process_ula(SM::ControlBlock& cb)
 	ue_ctxt->setSubscriptionStatus(ula_msg.subscription_status);
 	ue_ctxt->setNetAccessMode(ula_msg.net_access_mode);
 	ue_ctxt->setAccessRestrictionData(ula_msg.access_restriction_data);
-  ue_ctxt->setSubscribedApn(Apn_name(ula_msg.selected_apn));
-  log_msg(LOG_DEBUG, "Selected APN %s \n",ula_msg.selected_apn.val);
+	
+    ue_ctxt->setSubscribedApn(Apn_name(ula_msg.selected_apn));
+    log_msg(LOG_DEBUG, "Selected APN %s \n", ula_msg.selected_apn.val);
 
+    for (int i = 0; i < ula_msg.supp_features_list.count; i++)
+    {
+        if (ula_msg.supp_features_list.supp_features[i].feature_list_id == 2)
+        {
+            ue_ctxt->setHssFeatList2(
+                    ula_msg.supp_features_list.supp_features[i]);
+            break;
+        }
+    }
+        
+	ue_ctxt->setDcnrCapable(MmeCommonUtils::isUeNRCapable(*ue_ctxt));
+	
+	// UE-AMBR received from HSS. Treated as APN-AMBR as well.
+	// Bitrate values beyond 4.2 Gbps will be set in extended AMBR fields.
+	// Extended AMBR fields can store bit rate values upto 4.2 Tbps.
 	struct AMBR ambr;
 	ambr.max_requested_bw_dl = ula_msg.max_requested_bw_dl;
 	ambr.max_requested_bw_ul = ula_msg.max_requested_bw_ul;
-	
+	ambr.ext_max_requested_bw_dl = ula_msg.extended_max_requested_bw_dl;
+	ambr.ext_max_requested_bw_ul = ula_msg.extended_max_requested_bw_ul;
```

Comment:
do we know if values are always present ?  dcnr is configurable that means if MME does not send dcnr then MME will not get anything from HSS...

Reply:
Taken care in https://github.com/omec-project/Nucleus/pull/51/commits/d588d536035fba07e1b1ccddd992218318bf99d5